net/libfab/libfab.c

```diff
@@ -0,0 +1,2829 @@
+	if (rc != FI_SUCCESS)
+		return M0_RC(rc);
+
+	rc = libfab_waitfd_bind(&aep->aep_rx_res.frr_cq->fid, tm);
+	if (rc != FI_SUCCESS)
+		return M0_RC(rc);
+	
+	rc = fi_ep_bind(aep->aep_rxep, &tm->ftm_tx_cq->fid, FI_TRANSMIT);
+	if (rc != FI_SUCCESS)
+		return M0_RC(rc);
+	rc = fi_ep_bind(aep->aep_rxep, &aep->aep_rx_res.frr_cq->fid, FI_RECV);
+	if (rc != FI_SUCCESS)
+		return M0_RC(rc);
+
+	/* Initialise and bind event queue */
+	memset(&eq_attr, 0, sizeof(eq_attr));
+	eq_attr.wait_obj = FI_WAIT_FD;
+	eq_attr.size = FAB_MAX_AEP_EQ_EV;
+	rc = fi_eq_open(fab->fab_fab, &eq_attr, &aep->aep_rx_res.frr_eq, NULL);
+	if (rc != FI_SUCCESS)
+		return M0_RC(rc);
+
+	rc = libfab_waitfd_bind(&aep->aep_rx_res.frr_eq->fid, tm);
+	if (rc != FI_SUCCESS)
+		return M0_RC(rc);
+
+	rc = fi_ep_bind(aep->aep_rxep, &aep->aep_rx_res.frr_eq->fid, 0);
+	if (rc != FI_SUCCESS)
+		return M0_RC(rc);
+
+	/* Bind shared recv context */
+	rc = fi_ep_bind(aep->aep_rxep, &tm->ftm_rctx->fid, 0);
+	if (rc != FI_SUCCESS)
+		return M0_RC(rc);
+
+	return M0_RC(rc);
+}
+
+/**
+ * Initialize passive endpoint resources and associate
+ * it to the passive endpoint.
+ */
+static int libfab_pep_res_init(struct m0_fab__passive_ep *pep,
+			       struct m0_fab__tm *tm)
+{
+	struct fi_eq_attr eq_attr;
+	int               rc = 0;
+
+	/* Initialise and bind event queue */
+	memset(&eq_attr, 0, sizeof(eq_attr));
+	eq_attr.wait_obj = FI_WAIT_FD;
+	eq_attr.size = FAB_MAX_PEP_EQ_EV;
+	rc = fi_eq_open(tm->ftm_fab->fab_fab, &eq_attr, &pep->pep_res.fpr_eq,
+			NULL);
+	if (rc != FI_SUCCESS)
+		return M0_RC(rc);
+	
+	rc = libfab_waitfd_bind(&pep->pep_res.fpr_eq->fid, tm);
+	if (rc != FI_SUCCESS)
+		return M0_RC(rc);
+
+	rc = fi_pep_bind(pep->pep_pep, &pep->pep_res.fpr_eq->fid, 0);
+
+	return M0_RC(rc);
+}
+
+/**
+ * Used to accept an incoming connection request
+ */
+static int libfab_conn_accept(struct m0_fab__ep *ep, struct m0_fab__tm *tm,
+			      struct fi_info *info)
+{
+	struct m0_fab__active_ep *aep;
+	struct fid_domain        *dp;
+	int                       rc;
+
+	M0_ENTRY("from ep=%s", (char*)ep->fep_name.fen_str_addr);
+
+	aep = libfab_aep_get(ep);
+	dp = tm->ftm_fab->fab_dom;
+	
+	if (aep->aep_rxep != NULL) {
+		rc = fi_close(&aep->aep_rxep->fid);
+		if (rc != FI_SUCCESS)
+			M0_LOG(M0_ERROR, "ep close = %d",rc);
+		libfab_ep_rxres_free(&aep->aep_rx_res, tm);
+	}
+
+	rc = fi_endpoint(dp, info, &aep->aep_rxep, NULL);
+	if (rc != FI_SUCCESS) {
+		libfab_aep_param_free(aep, tm);
+		return M0_RC(rc);
+	}
+
+	rc = libfab_ep_rxres_init(aep, tm);
+	if (rc != FI_SUCCESS) {
+		libfab_aep_param_free(aep, tm);
+		return M0_RC(rc);
+	}
+
+	rc = fi_enable(aep->aep_rxep);
+	if (rc != FI_SUCCESS) {
+		libfab_aep_param_free(aep, tm);
+		return M0_RC(rc);
+	}
+
+	/* Accept incoming request */
+	rc = fi_accept(aep->aep_rxep, NULL, 0);
+	if (rc != FI_SUCCESS) {
+		libfab_aep_param_free(aep, tm);
+		return M0_RC(rc);
+	}
+
+	while (libfab_check_for_event(aep->aep_rx_res.frr_eq) != FI_CONNECTED);
```

Comment:
If this is a no-op loop, please write it as:
```
while (libfab_check_for_event(aep->aep_rx_res.frr_eq) != FI_CONNECTED)
        ;
```
Even we can add comments:
```
while (libfab_check_for_event(aep->aep_rx_res.frr_eq) != FI_CONNECTED)
        /* No-op */;
```

This will avoid unexpected bug.

Reply:
Corrected